hail_search/queries/mito.py

```diff
@@ -117,6 +124,236 @@ class MitoHailTableQuery(BaseHailTableQuery):
+        )
+        ht = ht.annotate(wes_passes_quality=wes_passes)
+        wgs_passes_quality_filter = self._get_family_passes_quality_filter(quality_filter, ht, WGS_FILTERS_FIELD, **kwargs)
+        wgs_passes = ht[WGS_FAMILY_ENTRIES_FIELD] if wgs_passes_quality_filter is None else (
+            ht[WGS_FAMILY_ENTRIES_FIELD].map(lambda entries: hl.or_missing(wgs_passes_quality_filter(entries), entries))
+        )
+        return ht.annotate(wgs_passes_quality=wgs_passes)
+
+    @staticmethod
+    def _build_family_index_map(sorted_wes_family_sample_data, sorted_wgs_family_sample_data):
+        family_guid_idx_map = defaultdict(dict)
+
+        for family_idx, samples in enumerate(sorted_wes_family_sample_data):
+            family_guid = samples[0]['familyGuid']
+            family_guid_idx_map[family_guid][SAMPLE_TYPE_WES] = family_idx
+
+        for family_idx, samples in enumerate(sorted_wgs_family_sample_data):
+            family_guid = samples[0]['familyGuid']
+            family_guid_idx_map[family_guid][SAMPLE_TYPE_WGS] = family_idx
+
+        return hl.dict(family_guid_idx_map)
+
+    def _apply_multi_sample_type_entry_filters(self, ht, family_idx_map) -> hl.Table:
+        # Keep family from both sample types if either passes quality AND inheritance
+        ht = ht.annotate(
+            wes_family_entries=hl.enumerate(ht[WES_FAMILY_ENTRIES_FIELD]).map(
+                lambda x: hl.or_missing(  # x[0] is family index, x[1] is list of samples
+                    self._family_passes_quality_inheritance(
+                        ht, SAMPLE_TYPE_WES, x[0], x[1][0]['familyGuid'], family_idx_map
+                    ), x[1]
+                )
+            ), wgs_family_entries=hl.enumerate(ht[WGS_FAMILY_ENTRIES_FIELD]).map(
+                lambda x: hl.or_missing(
+                    self._family_passes_quality_inheritance(
+                        ht, SAMPLE_TYPE_WGS, x[0], x[1][0]['familyGuid'], family_idx_map
+                    ), x[1]
+                )
+            )
+        )
+        # Filter out families with no valid entries in either sample type
+        ht = ht.filter(ht.wes_family_entries.any(hl.is_defined) | ht.wgs_family_entries.any(hl.is_defined))
+        return ht.transmute(
+            family_entries=hl.coalesce(
+                ht.wes_family_entries, hl.empty_array(ht.wes_family_entries.dtype.element_type)
+            ).extend(hl.coalesce(
+                ht.wgs_family_entries, hl.empty_array(ht.wgs_family_entries.dtype.element_type)
+            )).filter(lambda entries: entries.any(hl.is_defined)),
+            filters=hl.coalesce(ht.wes_filters, hl.empty_set(hl.tstr)).union(
+                hl.coalesce(ht.wgs_filters, hl.empty_set(hl.tstr))
+            )
+        )
+
+    @staticmethod
+    def _family_passes_quality_inheritance(ht, sample_type, sample_type_family_idx, family_guid, family_idx_map):
+        if sample_type == SAMPLE_TYPE_WES:
```

Comment:
I'm thinking about creating a SampleType enum that has these fields as properties instead of a big if/else block